packages/types/src/clerk.ts

```diff
@@ -525,14 +525,15 @@ export type BuildUrlWithAuthParams = {
 
 // TODO: Make sure Isomorphic Clerk navigate can work with the correct type:
 // (to: string) => Promise<unknown>
-export type CustomNavigation = (to: string) => Promise<unknown> | void;
+export type CustomNavigation = (to: string, options?: NavigateOptions) => Promise<unknown> | void;
 
 export type ClerkThemeOptions = DeepSnakeToCamel<DeepPartial<DisplayThemeJSON>>;
 
 export interface ClerkOptions {
   appearance?: Appearance;
   localization?: LocalizationResource;
-  navigate?: (to: string) => Promise<unknown> | unknown;
+  push?: (to: string) => Promise<unknown> | unknown;
+  replace?: (to: string) => Promise<unknown> | unknown;
```

Comment:
❓Shouldn't we also add a deprecation warning in v4 ?

Reply:
will do